mscore/pm.cpp

```diff
@@ -212,5 +249,49 @@ int PortMidiDriver::getDeviceIn(const QString& name)
             }
       return -1;
       }
+
+//---------------------------------------------------------
+//   getDeviceOut
+//---------------------------------------------------------
+
+int PortMidiDriver::getDeviceOut(const QString& name)
+      {
+      int interf = Pm_CountDevices();
+      for (int id = 0; id < interf; id++) {
+            const PmDeviceInfo* info = Pm_GetDeviceInfo((PmDeviceID)id);
+            if (info->output) {
+                  QString n = QString(info->interf) + "," + QString(info->name);
+                  if (n == name)
+                        return id;
+                  }
+            }
+      return -1;
+      }
+
+//---------------------------------------------------------
+//   isSameCoreMidiIacBus
+//    determines if both the input and output devices are the same shared CoreMIDI "IAC" bus
+//---------------------------------------------------------
+
+bool PortMidiDriver::isSameCoreMidiIacBus(const QString& inName, const QString& outName)
+      {
+      int interf = Pm_CountDevices();
+      const PmDeviceInfo* inInfo = 0;
+      const PmDeviceInfo* outInfo = 0;
+      for (PmDeviceID id = 0; id < interf; id++) {
+            const PmDeviceInfo* info = Pm_GetDeviceInfo((PmDeviceID)id);
+            if (info->input && inName == info->name)
+                  inInfo = info;
+            if (info->output && outName == info->name)
+                  outInfo = info;
+            }
+
+      if (inInfo && outInfo &&
+          QString(inInfo->interf) == "CoreMIDI" && QString(outInfo->interf) == "CoreMIDI" &&
+          inName.contains("IAC") && outName.contains("IAC"))
+            return true;
+      else
+            return false;
+      }
```

Comment:
this is a bit convoluted function...maybe it can be made simpler...I don't really know how much.  Unforutnatly I have to first search the list for the input and output infos, since don't already have their PmDeviceInfos directly.  I think I did those string compares correclty...note I don't know the best way to compare const char pointers with Qt, which is why I converted them to QString explicitly before using the ==